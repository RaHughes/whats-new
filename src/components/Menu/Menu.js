import React, { Component } from 'react';
import './Menu.scss'

class Menu extends Component {
  constructor() {
    super()
  }

  chooseNews = (event) => {
    event.preventDefault();
    this.props.pickNews(event.target.dataset.category)
  }

  render() {
    return (
      <div>
      <button data-category='local' onClick={event => this.chooseNews(event)}>Local News</button>
      <button data-category='technology' onClick={event => this.chooseNews(event)}>Technology</button>
      <button data-category='entertainment' onClick={event => this.chooseNews(event)}> Entertainment</button>
      <button data-category='science' onClick={event => this.chooseNews(event)}>Science</button>
      <button data-category='health' onClick={event => this.chooseNews(event)}>Health</button>
      </div>

    )
  }
}
export default Menu;