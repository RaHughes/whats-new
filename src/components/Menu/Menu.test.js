import React from 'react';
import { shallow } from 'enzyme';
import Menu from './Menu';
import { jsxEmptyExpression } from '@babel/types';

describe('Menu', ()=> { 
  it('renders without crashing', () => {
    const wrapper =
    <Menu 
    pickNews={jest.fn()}
    />

    expect(wrapper).toMatchSnapshot();
  });
});