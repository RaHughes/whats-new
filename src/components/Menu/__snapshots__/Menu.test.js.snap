// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu renders without crashing 1`] = `
<Menu
  pickNews={[MockFunction]}
/>
`;
